minicluster/src/main/java/alluxio/master/LocalAlluxioMaster.java

```diff
@@ -146,29 +142,29 @@ public boolean isServing() {
   }
 
   /**
-   * Stops the master and cleans up client connections.
+   * Stops the master processes and cleans up client connections.
    */
   public void stop() throws Exception {
-    // This shutdown needs to be done in a loop with retry because the interrupt signal can
-    // sometimes be ignored in the master implementation. For example, if the master is doing
-    // a hdfs listStatus RPC (hadoop version is 1.x), the interrupt signal is not properly handled.
-    while (mMasterThread.isAlive()) {
```

Comment:
I think this is part of greater discussion on how hard we should try to cleanup resources created a test cluster before shutting it down ... my take on this is that our tests should be written so that they are resilient to previous tests no being properly cleaned up (as it my be prohibitively expensive to do so), when it comes to leaking system resources, we should rely these resources being cleaned up when a JVM is shutdown

all in all, I think that the common case should be to terminate a test cluster by simply interrupting the server threads and use clean shutdown part way through a test with subsequent test logic relying on the clean shutdown